packages/hyperion-autologging/src/ALElementValuePublisher.ts

```diff
@@ -64,7 +64,7 @@ export function publish(options: InitOptions): void {
       reactComponentData = elementInfo.getReactComponentData();
     }
 
-    const elementText = getElementTextEvent(element, surface, tryInteractiveParentTextEventName, true);
+    const elementText = !disableElementTextExtraction ? getElementTextEvent(element, surface, tryInteractiveParentTextEventName, true) : { elementName: null, elementText: null };
```

Comment:
Ideally, the `getElementTextEvent` function itself should be reponsible to return the 'null' equivalent value. That way, we ensure it is the same all the time. 
So, we have two options:
- pass the flag directly to the function and let it decide to return something. It already has a code for `element`.
- pass a `source` or `origin`  to the function call and then only have the options `ALElementTextOptions` has this flag supported for various sources. 

I think the second options seems cleaner and doable (source would be either event names or 'surface'...

Reply:
Right now not passing an `element` argument to `getElementTextEvent` will return the null object as done here.  Should we do that instead?

I'm not sure I understand option 2,  ALElementTextOptions is global for all events,  how would we conditionally return something or nothing by event?

Reply:
yes, i think nulling the `element` would be better than hard coding the empty case.

Option 2 would look something like this:
* call the function with a `origin` argument which is `EventName | 'surface'`. Something like `getElementTextEvent(element, surface, ..., 'change')`
* in the `ALElementTextOptions` have a new flag like `disableTextExtractOrigins: ['surface', 'mouseover', ...]`.
* in the actual function, check the passed origin against the options. 

You don't have to do it this way, just an idea to consider.